src/DotNetty.Handlers/Tls/ClientTlsSettings.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Handlers.Tls
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Authentication;
+    using System.Security.Cryptography.X509Certificates;
+
+    public sealed class ClientTlsSettings : TlsSettings
+    {
+        IReadOnlyCollection<X509Certificate2> certificates;
+
+        public ClientTlsSettings(string targetHost)
+            : this(targetHost, new X509CertificateCollection())
+        {
+        }
+
+        public ClientTlsSettings(string targetHost, X509CertificateCollection certificates)
+            : this(false, certificates, targetHost)
+        {
+        }
+
+        public ClientTlsSettings(bool checkCertificateRevocation, X509CertificateCollection certificates, string targetHost)
+            : this(SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12, checkCertificateRevocation, certificates, targetHost)
+        {
+        }
+
+        public ClientTlsSettings(SslProtocols enabledProtocols, bool checkCertificateRevocation, X509CertificateCollection certificates, string targetHost)
```

Comment:
cert collection is mutable. How about accepting an array of certs instead and building collection either internally here or on authentication?